public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboard.tsx

```diff
@@ -27,29 +44,84 @@ const Loader = () => {
 };
 
 export const SharePublicDashboard = (props: Props) => {
-  const { data: publicDashboard, isLoading, isError } = useGetPublicDashboardQuery(props.dashboard.uid);
+  const { data: publicDashboard, isLoading: isGetLoading, isError } = useGetPublicDashboardQuery(props.dashboard.uid);
+  const [createPublicDashboard, { isLoading: isSaveLoading }] = useCreatePublicDashboardMutation();
+  const [update, { isLoading: isUpdateLoading }] = useUpdatePublicDashboardMutation();
+  const [deletePublicDashboard, { isLoading: isDeleteLoading }] = useDeletePublicDashboardMutation();
+  const isLoading = isGetLoading || isSaveLoading || isDeleteLoading;
+  const dashboardState = useSelector((store) => store.dashboard);
+  const dashboard = dashboardState.getModel()!;
+  const { unsupportedDataSources } = useGetUnsupportedDataSources(dashboard);
+  const timeRange = getTimeRange(dashboard.getDefaultTime(), dashboard);
+  const hasWritePermissions = contextSrv.hasPermission(AccessControlAction.DashboardsPublicWrite);
+  const hasTemplateVariables = dashboardHasTemplateVariables(dashboard.getVariables());
+
+  const onDeletePublicDashboardClick = (onDelete: () => void) => {
+    deletePublicDashboard({
+      dashboard,
+      uid: publicDashboard!.uid,
+      dashboardUid: dashboard.uid,
+    });
+    onDelete();
```

Comment:
this is what I mean. Many things that make this component quite complex and breaks the "encapsulation" or the duties between each component

Reply:
@juanicabanas I see your concern here, but we also need to have some temporary state where we make some compromises. Eventually, a few months from now the Scenes version will be the one that is the standard one and the encapsulation that you have in mind will be guaranteed by the scene object and scene object renderer separation.

But in the meantime we want to avoid re-implementing the UI you have worked so hard on, and re-use as much of the already existing code as possible. Also, the current encapsulation is IMO debatable, as the components there are mixing the logic (via hooks, and redux connection) and UI concerns, making them not re-usable in any other contexts.

Reply:
@dprokop 
yes, I agree with you about splitting to reuse the component. 
Scenes takes some values from one place, while now we're taking it from redux for example.

Also, you're right it is something debatable. I wonder if the update hook, for example, should be in the parent or inside the component itself, since that doesn't change either in non-scenes or scenes ecosystem and it avoids passing so many values as prop: `publicDashboard`, `update` function, `isLoading` related to update function and so on.
That's my point of view. Splitting logic and UI into 2 different components lets you reuse the whole UI without any problem but.. does the way of how we retrieve backend data change between those parents that render the same UI? Just a question and a little be concern about the big parent we would introduce right now.
Something to think: how likely is X component will need to be re-usable by another component? How much am I coupling those components?

Reply:
@juanicabanas I went to the code one more time and I think I see your concern better now, I think we were talking about the same things but haveing slightly different insights :D

There is s difference between scenes public dash tab and "old"(curent) public dash tab share tab implementation

In scenes implementation we have:

`ShareModal` -> `CreatePublicDashboard` -> **`CreatePublicDashboardComponent`** 

while in the "old" implementation the used components are 

`ShareModal` -> **`CreatePublicDashboard`**

where the "bold" highlighted components are the same. 

Do you mean that in the "old" implementation we should have another layer of components that encapsulate the api interactions? If so - i do agree with you. And thinking about this more, and given recent changes by @kaydelaney that introduce the typing of the dashboard in the RTK queries that now allow `DashboardModel | DashboardScene` it should be possible to have those components implement the API interactions instead of implementations in SharePublicDashboard old modal or the "middleware" `CreatePublicDashboard` component in the scenes version.

Reply:
Exactly. And in the end, it could be the same `**CreatePublicDashboard**` the one that has those hooks (since it's the same for scenes and the current implementation). 
Therefore, instead of passing down so many props to `CreatePublicDashboard` from a parent that, from my point of view, shouldn't have that logic (the share modal in this case), it would be just one component with everything wrapped. 
Otherwise, we should create a component in the middle of `ShareModal` and `CreatePublicDashboard`, as it is in scenes which.. I'm not pretty sure

Reply:
Good. @kaydelaney - is there anything that would prevent us from doing this? Maybe me and @juanicabanas are missing something that you have discovered while spending time on this.